File: C++/Longest Common Subsequence/insertion_deletion.cpp
#include<bits/stdc++.h>
using namespace std;

int lcs(string a, string b){
    int m = a.length();
    int n = b.length();
    int dp[n+1][m+1];

    for(int i=0; i<=m; i++){
        for(int j=0; j<=n; j++){
            if(i==0 || j==0){
                dp[i][j] = 0;
            } else if(a[i-1]==b[j-1]){
                dp[i][j] = 1 + dp[i-1][j-1];
            } else{
                dp[i][j] = max(dp[i-1][j], dp[i][j-1]);
            }
        }
    }
    return dp[m][n];
}

int insertdelete(string a, string b){
    int m = a.length();
    int n = b.length();
    int subseq = lcs(a, b);
    cout<<"Deletions: "<<max(m, n)-subseq<<'\n';
    cout<<"Insertions: "<<min(m, n)-subseq<<'\n';
}

int main(){
    int test;
    string a, b;
    cin>>test;
    while(test--){
        cin>>a;
        cin>>b;
        insertdelete(a, b);
    }
}

File: Python/Longest Common Subsequence/insertion_deletion.py
# Minimum number of Insertion and Deletion to make two strings equal
def lcs(a, b):
    m = len(a)
    n = len(b)
    dp = [[None for _ in range(n+1)] for _ in range(m+1)]

    for i in range(m+1):
        for j in range(n+1):
            if(i==0 or j==0):
                dp[i][j] = 0
            elif(a[i-1] == b[j-1]):
                dp[i][j] = dp[i-1][j-1] + 1
            else:
                dp[i][j] = max(dp[i-1][j], dp[i][j-1])
    return dp[m][n]

def insertdelete(a, b):
    subseq = lcs(a, b)
    m = len(a)
    n = len(b)
    print("Deletions: ", max(m, n)-subseq)
    print("Insertions: ", min(m, n)-subseq)

test = int(input())
for i in range(test):
    a = input()
    b = input()
    insertdelete(a, b)